actionview/test/template/url_helper_test.rb

```diff
@@ -50,6 +50,8 @@ class UrlHelperTest < ActiveSupport::TestCase
   include RenderERBUtils
 
   setup :_prepare_context
+  setup    { ActionView::Helpers::UrlHelper.button_to_generates_button_tag = true }
+  teardown { ActionView::Helpers::UrlHelper.button_to_generates_button_tag = false }
```

Comment:
@kaspth since the default is `false`, these lines were necessary to enable the new behavior.

Does this feel idiomatic and compatible with the new_framework_defaults pattern to you?

Currently, this commit only includes a single test to cover `button_to` without a block _with_ `button_to_generates_button_tag = false`. Is that sufficient? Would it be better to double the collection of `def test_button_to_*` tests to ensure we have coverage for both variations of the configuration flag?